sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CastSuite.scala

```diff
@@ -845,4 +846,11 @@ class CastSuite extends SparkFunSuite with ExpressionEvalHelper {
     val outputOuter = Row.fromSeq((1 to N).map(_ => outputInner))
     checkEvaluation(cast(Literal.create(inputOuter, fromOuter), toOuter), outputOuter)
   }
+
+  test("SPARK-22570: Cast should not create a lot of instance variables") {
```

Comment:
nit: instance variables -> global variables. To match with other two added tests.